### PR TITLE
Refactoring for bulk ingestion

### PR DESCRIPTION
**Pull request by aiwilliams**
The code was designed to limit ingestion to issues updated since the previous job execution time. The goal of this PR is to support `bulkIngest: true` at any time. A few bugs were discovered in the process of a careful code review and refactoring.

Another surprising (to me) discovery: issue ingestion limits are per-project. This means that were someone to add 15 projects, we would process 15*2000 issues on every execution.

### CODE REVIEW

**Comment by kevincasey1222** [comment]
FWIW from a non-official reviewer, all the code in this PR looks solid to me. Nice catches on the failure to ingest the `bulkIngest` property from local config, and on the failure to pass the page size to the `maxResults` property.

**Comment by kevincasey1222** [comment]
On the topic of the non-bulk Issue ingestion limit being per-project... I suppose we could protect the integration against unexpectedly long runs with another limit at the integration level, of say 10,000 Issues. Then five projects or less would be unaffected, but there would be some kind of guardrail against too many projects. You could even set the per-project actual limit to `10000/(max(numberOfProjects, 5))`

File: src/client.ts [hunk 1]
```diff
@@ -120,42 +126,95 @@ export class APIClient {
...
           pageSize: ISSUES_PAGE_SIZE,
         });
       } catch (err: any) {
         if (err.message.includes(`does not exist for the field 'project'.`)) {
-          this.logger.warn(
-            { projectKey },
-            'Project key does not exist or you do not have access to pull down issues from this project.',
+          this.logger.info(
+            { project: issuesOptions.project },
+            'Project key does not exist or permissions do not allow access to issues.',
```
Comment (ceelias) [inline src/client.ts:215]: This strikes me as something that should be exposed to the customer. Is there a reason why we wouldn't want to have this exposed as something that should be checked on the customer end?

File: .env.example [hunk 2]
```diff
@@ -1,4 +1,5 @@
 JIRA_HOST=somehost.com
 JIRA_USERNAME=someusername
 JIRA_PASSWORD=youknowwhatgoeshere
-PROJECTS=["Expecting","JSON","Array","of","projects"]
\ No newline at end of file
+PROJECTS=["Expecting","JSON","Array","of","projects"]
+BULK_INGEST=false
```
Comment (ceelias) [inline .env.example:5]: I'd be a fan of renaming this to be a bit more specific such as `BUILK_INGEST_ISSUES`

**Review by ceelias — APPROVED** [review]
Solid stuff in here!

File: src/client.ts (hunk 1 above)
Comment (aiwilliams) [reply src/client.ts:215]: Yah, that's a good point @ceelias. Created https://jupiterone.atlassian.net/browse/INT-2312 with acceptance criteria.

**Pull request merged**